### PR TITLE
fix(KFLUXBUGS-1215): prefetch: add git auth support and logging

### PR DESCRIPTION
**Pull request by scoheb**
- needed for private repos
- add ability to turn on cachi2 debug logging

fixes [KFLUXBUGS-1215](https://issues.redhat.com//browse/KFLUXBUGS-1215)

### CODE REVIEW

**Review by ralphbean — APPROVED** [review]

File: task/prefetch-dependencies/0.1/prefetch-dependencies.yaml [hunk 1]
```diff
@@ -55,29 +72,70 @@ spec:
...
+          echo "https://$(cat ${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/username):$(cat ${WORKSPACE_BASIC_AUTH_DIRECTORY_PATH}/password)@$HOSTNAME" > "${PARAM_USER_HOME}/.git-credentials"
+          echo -e "[credential \"https://$HOSTNAME\"]\n  helper = store" > "${PARAM_USER_HOME}/.gitconfig"
+        else
+          echo "Unknown basic-auth workspace format"
+          exit 1
+        fi
+        chmod 400 "${PARAM_USER_HOME}/.git-credentials"
+        chmod 400 "${PARAM_USER_HOME}/.gitconfig"
+        # needed or else you'll see "could not read Username for 'https://gitlab.com':"
+        cd $(workspaces.source.path)/source && git config remote.origin.url $(cat "${PARAM_USER_HOME}/.git-credentials")
```
Comment (chmeliik) [inline task/prefetch-dependencies/0.1/prefetch-dependencies.yaml:97]: This doesn't seem right, the clone task doesn't need it. 

I think this approach may also print credentials in the task logs in some cases (IIUC that's why https://github.com/redhat-appstudio/build-definitions/pull/861 was needed, unfortunately can't tell from the anymore from the description)

**Review by chmeliik — CHANGES_REQUESTED** [review]
I'm concerned with the amount of logic this copy-pastes from the clone task. Do we really need `# Compatibility with kubernetes.io/basic-auth secrets`? If not, this would just be a cp and chmod of two files.

If we do, then oh well. Let's call this a limitation of Tekton :smiling_face_with_tear:

File: task/prefetch-dependencies/0.1/prefetch-dependencies.yaml (hunk 1 above)
Comment (chmeliik) [reply task/prefetch-dependencies/0.1/prefetch-dependencies.yaml:97]: And cachi2 might include the credentials in the SBOM, it uses the origin url to construct the purl of packages coming from the source repo

Comment (chmeliik) [reply task/prefetch-dependencies/0.1/prefetch-dependencies.yaml:97]: > And cachi2 might include the credentials in the SBOM, it uses the origin url to construct the purl of packages coming from the source repo

Filed https://github.com/containerbuildsystem/cachi2/issues/521

File: task/prefetch-dependencies/0.1/README.md [hunk 2]
```diff
@@ -4,12 +4,14 @@ Task that uses Cachi2 to prefetch build dependencies.
...
 ## Parameters
-|name|description|default value|required|
-|---|---|---|---|
-|input|Configures project packages that will have their dependencies prefetched.||true|
-|dev-package-managers|Enable in-development package managers. WARNING: the behavior may change at any time without notice. Use at your own risk. |false|false|
+|name| description                                                                                                                                         |default value|required|
+|---|-----------------------------------------------------------------------------------------------------------------------------------------------------|---|---|
+|input| Configures project packages that will have their dependencies prefetched.                                                                           ||true|
+|dev-package-managers| Enable in-development package managers. WARNING: the behavior may change at any time without notice. Use at your own risk.                          |false|false|
+|enable-debug-logging| Enable debug logging with cachi2 |false|false|
```
[the remark this answers is not in the document]
Comment (scoheb) [reply task/prefetch-dependencies/0.1/README.md]: done

File: task/prefetch-dependencies/0.1/prefetch-dependencies.yaml (hunk 1 above)
Comment (scoheb) [reply task/prefetch-dependencies/0.1/prefetch-dependencies.yaml]: I have altered my approach. please review

File: task/prefetch-dependencies/0.1/prefetch-dependencies.yaml [hunk 3]
```diff
@@ -38,6 +47,14 @@ spec:
...
     - name: DEV_PACKAGE_MANAGERS
       value: $(params.dev-package-managers)
+    - name: ENABLE_DEBUG
+      value: $(params.enable-debug-logging)
+    - name: WORKSPACE_BASIC_AUTH_DIRECTORY_BOUND
+      value: $(workspaces.basic-auth.bound)
+    - name: WORKSPACE_BASIC_AUTH_DIRECTORY_PATH
+      value: $(workspaces.basic-auth.path)
+    - name: PARAM_USER_HOME
+      value: $(params.userHome)
```
[the remark this answers is not in the document]
Comment (scoheb) [reply task/prefetch-dependencies/0.1/prefetch-dependencies.yaml:57]: not sure this is relevant anymore based on my new approach

**Comment by chmeliik** [comment]
> There's no need to do this differently than the clone task. I've tested it and opened #953

If that looks reasonable, let's keep only the logging option here

**Comment by scoheb** [comment]
Closed in favour of #953

**Pull request closed without merging**